examples/secrets/secrets.go

```diff
@@ -14,115 +14,181 @@
  * limitations under the License.
  */
 
+// Package secrets provides a utility to read pre-created or dynamically create keys and certificates for Things.
+// Only intended for examples and demos NOT production use.
 package secrets
 
 import (
 	"crypto"
 	"crypto/ecdsa"
+	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/x509"
 	"crypto/x509/pkix"
 	"encoding/json"
-	"log"
+	"fmt"
+	"io/ioutil"
 	"math/big"
+	"os"
 	"time"
 
 	"gopkg.in/square/go-jose.v2"
 )
 
-var keyStore = "{\"keys\":[" +
-	"{\"use\":\"sig\"," +
-	"\"kty\":\"EC\"," +
-	"\"kid\":\"572ddcde-1532-4175-861b-0622ac2f3bf3\"," +
-	"\"crv\":\"P-256\"," +
-	"\"alg\":\"ES256\"," +
-	"\"x\":\"J4myf5y8cpKubDrit6RLnF3FAf__VMjZzdIFMF9yv3M\"," +
-	"\"y\":\"6eX5cj0oVhiLSpzTgvqtZHjp7IBr4aIO5Jl_KfNnq4c\"," +
-	"\"d\":\"jyWleC86wyd0PppZTySEb8NK7AtaJe9Xol8uqQ0gBZQ\"}," +
-	"{\"use\":\"sig\"," +
-	"\"kty\":\"EC\"," +
-	"\"kid\":\"d21f083f-c6a1-4ae7-94a4-e2604bb25e1d\"," +
-	"\"crv\":\"P-256\"," +
-	"\"alg\":\"ES256\"," +
-	"\"x\":\"05-ue0KZ15MAoR4iSkPXePLBpelIFwNy4yWQF-1MLSA\"," +
-	"\"y\":\"v4kX8AhvY990LZLsIfwaPyKsGCsqAc05k5-9Z325hB4\"," +
-	"\"d\":\"2bX0-tNfO2fwTANLUeje9_NoF1IETmMqJsEjBw6Pens\"}," +
-	"{\"use\":\"sig\"," +
-	"\"kty\":\"EC\"," +
-	"\"kid\":\"47cf707c-80c1-4816-b067-99db2a443113\"," +
-	"\"crv\":\"P-256\"," +
-	"\"alg\":\"ES256\"," +
-	"\"x\":\"6cS0SXSsILpyzSi6DgLL2qKZIRsgkmUaus6qfzWl7Ek\"," +
-	"\"y\":\"x-tUNJOcD2AVULh19d1eLNsjCqIX-uHJ0cWdw_WPZss\"," +
-	"\"d\":\"EgRndTr2jv4rv_rAHPiziL-Q8nm523cZCwS5UPIMGVk\"}," +
-	"{\"use\":\"sig\"," +
-	"\"kty\":\"EC\"," +
-	"\"kid\":\"9c753fa7-68a3-418a-9571-e0eda88b1617\"," +
-	"\"crv\":\"P-256\"," +
-	"\"alg\":\"ES256\"," +
-	"\"x\":\"KCCUz_tv2swnOwytUCi6yIbyx22w_xjqWAypob61vJE\"," +
-	"\"y\":\"RT8XY1HCIso3HoUif9oBpwee2CBfjouqf2JjtFT6lKs\"," +
-	"\"d\":\"Fulx_Vjc-jRmYP_0K6TVFoIuK9MEYOgHCXRxCe3LZkY\"}]}"
+const ec256Test = `{"kty": "EC",
+	"kid": "Fol7IpdKeLZmzKtCEgi1LDhSIzM=",
+    "x": "N7MtObVf92FJTwYvY2ZvTVT3rgZp7a7XDtzT_9Rw7IA",
+	"y": "uxNmyoocPopYh4k1FCc41yuJZVohxlhMo3KTIJVTP3c",
+	"crv": "P-256",
+	"alg": "ES256",
+	"d": "w9rAMaNcP7cA0e5SECc4Tk1PDQEY66ml9y9-6E8fmR4",
+	"x5c": ["MIIBwjCCAWkCCQCw3GyPBTSiGzAJBgcqhkjOPQQBMGoxCzAJBgNVBAYTAlVLMRAwDgYDVQQIEwdCcmlzdG9sMRAwDgYDVQQHEwdCcmlzdG9sMRIwEAYDVQQKEwlGb3JnZVJvY2sxDzANBgNVBAsTBk9wZW5BTTESMBAGA1UEAxMJZXMyNTZ0ZXN0MB4XDTE3MDIwMzA5MzQ0NloXDTIwMTAzMDA5MzQ0NlowajELMAkGA1UEBhMCVUsxEDAOBgNVBAgTB0JyaXN0b2wxEDAOBgNVBAcTB0JyaXN0b2wxEjAQBgNVBAoTCUZvcmdlUm9jazEPMA0GA1UECxMGT3BlbkFNMRIwEAYDVQQDEwllczI1NnRlc3QwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQ3sy05tV/3YUlPBi9jZm9NVPeuBmntrtcO3NP/1HDsgLsTZsqKHD6KWIeJNRQnONcriWVaIcZYTKNykyCVUz93MAkGByqGSM49BAEDSAAwRQIgZhTox7WpCb9krZMyHfgCzHwfu0FVqaJsO2Nl2ArhCX0CIQC5GgWD5jjCRlIWSEFSDo4DZgoQFXaQkJUSUbJZYpi9dA=="]
```

Comment:
If I dynamically create the CA certificate, thing auth fails with:

`Path does not chain with any of the trust anchors`